near-sdk/src/store/lookup_map/mod.rs

```diff
@@ -190,19 +154,21 @@ where
         Q: BorshSerialize,
         K: Borrow<Q>,
     {
-        // Concat the prefix with serialized key and hash the bytes for the lookup key.
-        buffer.extend(prefix);
+        // Hash the serialized key
         key.serialize(buffer).unwrap_or_else(|_| env::panic_str(ERR_ELEMENT_SERIALIZATION));
 
-        H::hash(buffer)
+        // Prefix the hash for lookup
+        let mut key = Vec::with_capacity(buffer.len() + prefix.len());
```

Comment:
I'm confused a little by these comments. Is it still doing hashing somewhere? Or is one of the prefix, key or buffer expected to be already hashed?